### DISCUSSION TITLE
キャラクター・スタイルのプリセット追加に関して

### DISCUSSION THREAD

**Discussion by KAZU-Gamer** [ciffelia/koe#74] [Q&A] [answered]
VOICEVOX　ENGINEに入っているその他のスタイル等を追加する場合、「voicevox_presets.yaml」に記載する形で問題無いとは思いますが、追加できる数に制限等はあるのでしょうか？
もしくは、設定が必要な箇所が他にもあるのでしょうか？
異なるボイス単体で入れた場合には問題なく動作しましたが、記載数（ファイルに記載した数）が25個になった場合には、/voice　コマンド実行時にエラーが発生します・

**Comment by ciffelia** [comment]
追加できる数に制限は設けていないためバグの可能性があります。後日詳しく検証しますが、よろしければお使いの`voicevox_presets.yaml`をここで共有していただけますか？

**Reply by KAZU-Gamer** [reply]
ご連絡ありがとうございます。
あの後、エラー内容を確認した所、DiscordAPI側の制限？のように思われました。
Docker上で記載されたエラーログを含めお送りいたします。
[送付用.zip](https://github.com/ciffelia/koe/files/11074618/default.zip)

**Reply by ciffelia** [reply]
確認したところ、Discord API側で選択項目の上限が設定されているようです。現在の方式では25個以上表示させることはできないため、別の方法の追加を検討します。ご報告ありがとうございました。

**Comment by ciffelia** [comment, marked as answer]
[v2.0.0](https://github.com/ciffelia/koe/releases/tag/v2.0.0)で修正しました。
